src/parser/mod.rs

```diff
@@ -708,13 +708,16 @@ impl<'a> Parser<'a> {
             };
         };
 
+        let on_cluster = self.parse_optional_on_cluster()?;
```

Comment:
ON CLUSTER is a ClickHouse-specific syntax, but we don't add the dialect guard for the sake of keeping consistent with other places.